modulus/models/dlwp_healpix/HEALPixRecUNet.py

```diff
@@ -0,0 +1,471 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from dataclasses import dataclass
+from typing import Sequence
+
+import pandas as pd
+import torch as th
+from hydra.utils import instantiate
+from omegaconf import DictConfig
+
+from modulus.models.dlwp_healpix_layers import HEALPixFoldFaces, HEALPixUnfoldFaces
+from modulus.models.meta import ModelMetaData
+from modulus.models.module import Module
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class MetaData(ModelMetaData):
+    """Metadata for the DLWP HEALPix Model"""
+
+    name: str = "DLWP_HEALPixRec"
+    # Optimization
+    jit: bool = False
+    cuda_graphs: bool = True
+    amp_cpu: bool = True
+    amp_gpu: bool = True
+    # Inference
+    onnx: bool = False
+    # Physics informed
+    var_dim: int = 1
+    func_torch: bool = False
+    auto_grad: bool = False
+
+
+class HEALPixRecUNet(Module):
+    """Deep Learning Weather Prediction (DLWP) recurrent UNet model on the HEALPix mesh."""
+
+    def __init__(
+        self,
+        encoder: DictConfig,
```

Comment:
One potential issue is that this encoder config will not be json serializable. Not sure though. We might need to convert it to a normal python dict if this is the case.